File: 0009-palindrome-number/0009-palindrome-number.cs
public class Solution {
    public bool IsPalindrome(int x) {

            int[] arr = x
            .ToString()
            .Select(c => c - '0')
            .ToArray();

            int s = 0;
            int e = arr.Length - 1;
            while(s < e){
                if(arr[s]!= arr[e]){
                    return false;
                }
                s++;
                e--;
            }
            return true;

    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
using namespace std;
class Solution {
public:
    bool isPalindrome(int x) {
        vector<int>arr;
        string s = to_string(x);
        for(char c : s){
            arr.push_back(c - '0');
        }
        int start = 0;
        int end = arr.size() -1;
        while(start < end){
        if(arr[start] != arr[end]){
            return false;
        }
        start++;
        end--;
        }
        return true;

    }
};